File: mytime-api/app/models/register_user.py
from pydantic import BaseModel, EmailStr, Field
from typing import Optional


class RegisterUserModel(BaseModel):
    id: Optional[int] = None
    employee_id: Optional[int] = None

    first_name: str
    last_name: str
    email: EmailStr
    phone: str

    department_id: Optional[int] = None
    role_id: Optional[int] = None

    password: Optional[str] = Field(None, min_length=8)

    class Config:
        orm_mode = True

File: src/app/admin/models/register_user.ts
export interface RegisterUser {
    id?: number;
    password?: string;
    employee_id?: number;
    first_name?: string;
    last_name?: string;
    role_id?: number;
    department_id?: number;
    email?: string;
    phone?: string;
    is_active?: boolean;
    created_by?: number;
}